platform_google.py

```diff
@@ -43,6 +47,115 @@ def create_service():
     service = build('androidpublisher', 'v3', credentials=credentials)
     return service
 
+
+def get_subscription_info(package_name: str, product_id: str, err: base.ErrorSink):
+    service = create_service()
+    result = None
+    try:
+        response = service.monetization().subscriptions().get(
+            packageName=package_name,
+            productId=product_id
+        ).execute()
+
+        result = response
+    except Exception as e:
+        err.msg_list.append(f'Failed to get subscription details: {e}')
+
+    # assert result is None if err.has() else isinstance(result, SubscriptionV2Data)
+    return result
+
+
+def get_subscription_details_for_product_id(package_name: str, product_id: str, err: base.ErrorSink) -> dict[str, SubscriptionProductDetails] | None:
+    result = None
+
+    subscriptions = get_subscription_info(package_name, product_id, err)
+    if subscriptions is None:
+        return result
+
+    base_plans = json_dict_require_array(subscriptions, "basePlans", err)
+
+    for plan in base_plans:
+        base_plan_id = json_dict_require_str(plan, "basePlanId", err)
+        auto_renewing_base_plan_type = json_dict_optional_obj(plan, "autoRenewingBasePlanType", err)
+        grace_period_duration = json_dict_require_str(auto_renewing_base_plan_type, "gracePeriodDuration", err)
+        billing_period_duration = json_dict_require_str(auto_renewing_base_plan_type, "billingPeriodDuration", err)
+
+        if grace_period_duration[0] != "P":
+            err.msg_list.append(f'Grace period duration must be ISO 8601 format but does not start with "P" ({grace_period_duration[0]})')
+        if grace_period_duration[-1] != "D":
+            err.msg_list.append(f'Grace period duration must be ISO 8601 format but does not end with "D" ({grace_period_duration[-0]})')
```

Comment:
Log line is printing `[-0]`, meant to be `[-1]`?

Would be nice to have an example of the payload in a comment so that we can desk check the implementation since we're doing a bunch of string manipulation here.